backend/src/api/routers/projects.py

```diff
@@ -0,0 +1,138 @@
+import uuid
+from typing import Annotated, AsyncIterator
+
+import pydantic
+from api.auth.users import current_active_user
+from database.session import async_session_maker
+from fastapi import APIRouter, Depends, HTTPException, status
+from models.project import Project
+from sqlalchemy import select
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+
+router = APIRouter()
+
+
+class ReadProject(pydantic.BaseModel):
+    id: uuid.UUID
+    title: str
+    summary: str | None = None
+    status: str
+
+
+class CreateProject(pydantic.BaseModel):
+    title: str
+    summary: str | None = None
+    status: str
+
+
+class UpdateProject(pydantic.BaseModel):
+    title: str | None = None
+    summary: str | None = None
+    status: str | None = None
+
+
+async def get_project(project_id: uuid.UUID, session: AsyncSession) -> Project:
+    try:
+        project_entry = (
+            await session.execute(select(Project).filter_by(id=project_id))
+        ).scalar_one()
+    except NoResultFound:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND)
+    return project_entry
+
+
+async def transactional_session() -> AsyncIterator[AsyncSession]:
+    async with async_session_maker.begin() as session:
+        yield session
```

Comment:
Should we move the function to the database folder? This will probably be reused in every router.

Reply:
Absolutely I'll move it before merging.